app/services/proxy_load_balancer.rb

```diff
@@ -0,0 +1,28 @@
+class ProxyLoadBalancer
+  def initialize
+    all_proxies = SmartProxy.unscoped.with_features(RemoteExecutionProvider.providers.keys)
+    @jobs = get_counts(all_proxies)
+  end
+
+  # Get the least loaded proxy from the given list of proxies
+  def next(proxies)
+    ids = proxies.map(&:id)
```

Comment:
You can use the proxy objects directly in `@jobs` keys, which allows us to avoid loading the proxy object again for every host iteration